alchemist-loading/src/main/kotlin/it/unibo/alchemist/loader/m2m/SimulationModel.kt

```diff
@@ -407,12 +422,34 @@ internal object SimulationModel {
             root is Map<*, *> && DocumentRoot.Export.Data.validateDescriptor(root) -> {
                 val molecule = root[DocumentRoot.Export.Data.molecule]?.toString()
                 if (molecule == null) {
-                    visitBuilding(context, root)
+                    visitBuilding<Extractor<E>>(context, root)
                 } else {
                     val property = root[DocumentRoot.Export.Data.property]?.toString()
                     val filter: FilteringPolicy = root[DocumentRoot.Export.Data.valueFilter]
                         ?.let { CommonFilters.fromString(it.toString()) }
                         ?: CommonFilters.NOFILTER.filteringPolicy
+                    val precision: Int? = when (val digits = root[DocumentRoot.Export.Data.precision]) {
+                        null -> null
+                        is Int -> digits
```

Comment:
I dunno if it makes sense, but I would prefer something like this:
```
null, is Int -> digits
```
Why you need two separated cases?

Reply:
Yes, nice, please PR me with that!